lib/ui/tabs/home_widget.dart

```diff
@@ -197,14 +200,30 @@ class _HomeWidgetState extends State<HomeWidget> {
     UpdateService.instance.shouldUpdate().then((shouldUpdate) {
       if (shouldUpdate) {
         Future.delayed(Duration.zero, () {
-          showDialog(
-            context: context,
-            builder: (BuildContext context) {
-              return AppUpdateDialog(
-                UpdateService.instance.getLatestVersionInfo(),
-              );
+          SharedPreferences.getInstance().then(
+            (value) {
+              final lastDialogShownTime =
```

Comment:
Maybe refactor and reuse the code from here?

https://github.com/ente-io/photos-app/blob/d27a70bd37620ab925264f13b1487ff7b2891d46/lib/services/update_service.dart#L76